✨(learninglocker) configure and use pm2 to run learninglocker

lunika opened the pull request:
## Purpose

Recent docker images of learninglocker have [PM2](https://pm2.keymetrics.io/) installed.

We want to use this process manager in Arnold. For each service we created a configuration file containing information on how to run each application.

Also PM2 writes files and directories to the user's HOME, but by default in OpenShift, the HOME environment variable is set to `/` which is obviously not writable for an un-privileged user.

We need to set this variable to a directory the user can write to.

## Proposal

- [x] create a configuration file for each service
- [x] change DC command to run `pm2-runtime`

lunika committed:
✨(learninglocker) configure and use pm2 to run learninglocker

Recent docker images of learninglocker have pm2 installed. We want to
use this process manager in Arnold. For each service we created a
configuration file containing information on how to run each
application.
Also pm2 writes files and direcotries in the HOME user but by default on
OpenShift the HOME environment variable is set to `/` which is obviously
not writable for an un-privileged user. We need to set this variable
with a directory the user can writes in.

sampaccoud commented on apps/learninglocker/templates/services/app/_dc_base.yml.j2:
```diff
@@ -41,10 +41,26 @@ spec:
+            # PM2 creates files and directory in the user HOME. By default the user HOME is `/`
+            # and obviously it's not possible for an un-previleged to write in `/` directory.
+            # By changing the `HOME` environment variable with a directory the user can wrtie in
+            # PM2 will work as expected.
```
```suggestion
            # PM2 creates files and directories in the user's HOME. By default the user's HOME is `/`
            # and obviously it's not possible for an unprivileged user to write to the `/` directory.
            # PM2 will work as expected after changing the `HOME` environment variable to a
            # directory the user can write to.
```

sampaccoud commented on apps/learninglocker/templates/services/app/configs/ecosystem.config.js.j2:
```diff
@@ -0,0 +1,19 @@
+module.exports = [{
```
curious about the name of the file containing `ecosystem` and not 'pm2`.

lunika replied:
I took the name in the documentation https://pm2.keymetrics.io/docs/usage/application-declaration/

lunika committed:
🔧(learninglocker) add affinity rules on learninglocker pods

Affinity rules were missing in DeploymentConfig templates. This is
important to add them to not have all pods spawn on a same node.

jmaupetit commented on CHANGELOG.md:
```diff
@@ -13,6 +13,7 @@ Versioning](http://semver.org/spec/v2.0.0.html).
 - New `nextcloud` application
 - Add nginx status endpoint in `ashley`, `edxapp`, `edxec`, `flower`,
   `learninglocker`, `marsha`, `nextcloud` and `richie`.
+- Run learninglocker using pm2 process manager
```
Should be in the `changed` section instead.

jmaupetit commented on apps/learninglocker/templates/services/app/_dc_base.yml.j2:
```diff
@@ -41,10 +54,26 @@ spec:
             - name: learninglocker-v-storage
               mountPath: /app/storage
+            - name: learninglocker-configmap
+              mountPath: /pm2_config
```
The following is more UNIX-friendly IMO.

```suggestion
              mountPath: /usr/local/etc/pm2/config.d
```
